asm/custom-functions/src/lib.rs

```diff
@@ -190,19 +192,75 @@ pub fn process_startflags() {
             }
         }
     }
+
+    let additional_start_options = flag_mem.next_u16().unwrap_or_default();
+
+    // Starting rupee count.
+    // Last 7 bits.
     itemflag_set_to_value(
         501, /* rupee counter */
-        flag_mem.next_u16().unwrap_or_default(),
+        (additional_start_options & 0x7F) * 100,
     );
-    // heart capacity
-    let health_capacity = flag_mem.next_u8().unwrap_or_default();
+
+    // Starting heart capacity.
+    // Next 5 bits.
+    let health_capacity = (additional_start_options >> 7 & 0x1F) * 4;
     unsafe { (*FILE_MANAGER).FA.health_capacity = health_capacity.into() };
     unsafe { (*FILE_MANAGER).FA.current_health = health_capacity.into() };
-    // starting interface choice
-    let mode = flag_mem.next_u8().unwrap_or_default();
-    storyflag_set_to_value(840, mode.into());
 
-    // commit global flag managers
+    // Starting interface choice.
+    // Next 2 bits.
+    let interface = additional_start_options >> 12 & 0x3;
+    storyflag_set_to_value(840, interface.into());
+
+    // Starting bugs.
+    // Next 1 bit.
+    if additional_start_options >> 14 & 0x1 == 1 {
+        let mut counter = 10;
+        while counter <= 21 {
+            unsafe {
+                increaseCounter(counter, 99);
+                setFlagForItem(counter + 131); // counter + (itemflag - counter)
+            }
+            counter += 1;
+        }
+    }
+
+    // Starting treasures.
+    // First 1 bit.
+    if additional_start_options >> 15 & 0x1 == 1 {
+        let mut counter = 22;
+        while counter <= 37 {
+            unsafe {
+                increaseCounter(counter, 99);
+                setFlagForItem(counter + 139); // counter + (itemflag - counter)
+            }
+            counter += 1;
+        }
+    }
+
+    let additional_start_options_2 = flag_mem.next_u8().unwrap_or_default();
+
+    let mut pouch_items: [i32; 8] = [0; 8];
+    let mut next_pouch_index = 0usize;
+
+    // Starting Hylian Shield.
+    // 4th bit.
+    if additional_start_options_2 >> 3 & 0x1 == 1 {
+        pouch_items[next_pouch_index] = 125 | 0x30 << 0x10; // ID for Hylian Shield + durability
+        next_pouch_index += 1;
+    }
+
+    let mut bottles_added = 0;
+    while bottles_added < additional_start_options_2 & 0x7 {
+        pouch_items[next_pouch_index] = 153; // ID for bottles
+        bottles_added += 1;
+        next_pouch_index += 1;
+    }
```

Comment:
```suggestion
    for _bottles_added in 0..(additional_start_options_2 & 0x7) {
        pouch_items[next_pouch_index] = 153; // ID for bottles
        next_pouch_index += 1;
    }
```
I'll look at this again, this array access probably creates an unnecessary panicking branch, since the index can be out of bounds